### PR TITLE
Validation of prescriptions

### PR DESCRIPTION
**Pull request by ofrancon**
Similar to validation of predictions, added a `prescriptor_validation.py` to check the output produced by prescriptors.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,6 @@ examples/predictors/lstm/models/
 examples/predictors/lstm/predictions/
 examples/predictors/lstm/tests/fixtures/OxCGRT_latest.csv
 
+examples/prescriptors/zero/prescriptions/
```
Comment (ofrancon) [inline .gitignore:19]: I only validated `zero` so far, and put its prescriptions in this git ignored folder

File: examples/prescriptors/zero/prescribe.py [hunk 2]
```diff
@@ -15,7 +15,8 @@
             'C8_International travel controls',
             'H1_Public information campaigns',
             'H2_Testing policy',
-            'H3_Contact tracing']
+            'H3_Contact tracing',
+            'H6_Facial Coverings']
```
Comment (ofrancon) [inline examples/prescriptors/zero/prescribe.py:19]: `zero` was missing the last NPI for masks

File: examples/prescriptors/zero/prescribe.py [hunk 3]
```diff
@@ -28,6 +29,7 @@ def prescribe(start_date_str: str,
     hdf = pd.read_csv(path_to_hist_file,
                       parse_dates=['Date'],
                       encoding="ISO-8859-1",
+                      dtype={"RegionName": str},
```
Comment (ofrancon) [inline examples/prescriptors/zero/prescribe.py:32]: Suppresses a Pandas warning

File: validation/prescriptor_validation.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 (c) Cognizant Digital Business, Evolutionary AI. All rights reserved. Issued under the Apache 2.0 License.
+
+from typing import List
+
+import pandas as pd
+
+from validation.scenario_generator import ID_COLS, NPI_COLUMNS
+from validation.predictor_validation import _check_columns, _check_geos, _check_days
```
Comment (ofrancon) [inline validation/prescriptor_validation.py:8]: Reused the predictor_validation's code. Might need some refactoring.

File: validation/prescriptor_validation.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+def _check_prescription_values(df):
+    # For each IP column, check the values are valid
+    errors = []
+    for ip_name, ip_max_value in IP_MAX_VALUES.items():
+        if df[ip_name].isnull().values.any():
+            errors.append(f"Column {ip_name} contains NaN values")
+        if df[ip_name].min() < 0:
+            errors.append(f"Column {ip_name} contains negative values")
+        if df[ip_name].max() > ip_max_value:
+            errors.append(f"Column {ip_name} contains values higher than max possible value")
```
Comment (ofrancon) [inline validation/prescriptor_validation.py:76]: That's the main difference with predictions: here we check the values of the IP columns, min and max and NaN

File: validation/tests/fixtures/prescriptor/bad_dates_submission.csv [hunk 6]
```diff
@@ -0,0 +1,15 @@
+CountryName,RegionName,Date,C1_School closing,C2_Workplace closing,C3_Cancel public events,C4_Restrictions on gatherings,C5_Close public transport,C6_Stay at home requirements,C7_Restrictions on internal movement,C8_International travel controls,H1_Public information campaigns,H2_Testing policy,H3_Contact tracing,H6_Facial Coverings,PrescriptionIndex
```
Comment (ofrancon) [inline validation/tests/fixtures/prescriptor/bad_dates_submission.csv:1]: The old `fixtures` files have moved to a `predictor` subfolder. These new files are in a `prescriptor` subfolder

File: validation/tests/test_predictor_validation.py [hunk 7]
```diff
@@ -7,18 +7,20 @@
 
 ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
 FIXTURES_PATH = os.path.join(ROOT_DIR, "fixtures")
+PREDS_PATH = os.path.join(FIXTURES_PATH, "predictor")
```
Comment (ofrancon) [inline validation/tests/test_predictor_validation.py:10]: The `predictor` output files to test are now in a sub directory

File: examples/prescriptors/zero/prescribe.py (hunk 2 above)
Comment (dsargent) [reply examples/prescriptors/zero/prescribe.py:19]: Can we have the NPIs in a single file in this library so they can be imported from anywhere? Even more useful now this library is pip-installable.

**Review by dsargent — APPROVED** [review]
LGTM

File: examples/prescriptors/zero/prescribe.py (hunk 2 above)
Comment (ofrancon) [reply examples/prescriptors/zero/prescribe.py:19]: We can probably refactor that, yes

**Pull request merged**